src/test/regress/sql/failure_savepoints.sql

```diff
@@ -0,0 +1,190 @@
+SELECT citus.mitmproxy('conn.allow()');
+
+SET citus.shard_count = 2;
+SET citus.shard_replication_factor = 1; -- one shard per worker
+SET citus.next_shard_id TO 100950;
+ALTER SEQUENCE pg_catalog.pg_dist_placement_placementid_seq RESTART 150;
+
+CREATE TABLE artists (
+    id bigint NOT NULL,
+    name text NOT NULL
+);
+SELECT create_distributed_table('artists', 'id');
+
+-- add some data
+INSERT INTO artists VALUES (1, 'Pablo Picasso');
+INSERT INTO artists VALUES (2, 'Vincent van Gogh');
+INSERT INTO artists VALUES (3, 'Claude Monet');
+INSERT INTO artists VALUES (4, 'William Kurelek');
+
+SELECT citus.clear_network_traffic();
+
+-- simply fail at SAVEPOINT
+SELECT citus.mitmproxy('conn.onQuery(query="^SAVEPOINT").kill()');
+
+BEGIN;
+INSERT INTO artists VALUES (5, 'Asher Lev');
+SAVEPOINT s1;
+DELETE FROM artists WHERE id=4;
+RELEASE SAVEPOINT s1;
+COMMIT;
+
+SELECT * FROM artists WHERE id IN (4, 5);
+
+-- fail at RELEASE
+SELECT citus.mitmproxy('conn.onQuery(query="^RELEASE").kill()');
+
+BEGIN;
+INSERT INTO artists VALUES (5, 'Asher Lev');
+SAVEPOINT s1;
+DELETE FROM artists WHERE id=4;
+RELEASE SAVEPOINT s1;
+COMMIT;
+
+SELECT * FROM artists WHERE id IN (4, 5);
+
+-- fail at ROLLBACK
+SELECT citus.mitmproxy('conn.onQuery(query="^ROLLBACK").kill()');
+
+BEGIN;
+INSERT INTO artists VALUES (5, 'Asher Lev');
+SAVEPOINT s1;
+DELETE FROM artists WHERE id=4;
+ROLLBACK TO SAVEPOINT s1;
+COMMIT;
+
+SELECT * FROM artists WHERE id IN (4, 5);
+
+-- fail at second RELEASE
+SELECT citus.mitmproxy('conn.onQuery(query="^RELEASE").after(1).kill()');
+BEGIN;
+SAVEPOINT s1;
+DELETE FROM artists WHERE id=4;
+RELEASE SAVEPOINT s1;
+SAVEPOINT s2;
+INSERT INTO artists VALUES (5, 'Jacob Kahn');
+RELEASE SAVEPOINT s2;
+COMMIT;
+
+SELECT * FROM artists WHERE id IN (4, 5);
+
+-- fail at second ROLLBACK
+SELECT citus.mitmproxy('conn.onQuery(query="^ROLLBACK").after(1).kill()');
+
+BEGIN;
+SAVEPOINT s1;
+UPDATE artists SET name='A' WHERE id=4;
+ROLLBACK TO SAVEPOINT s1;
+SAVEPOINT s2;
+DELETE FROM artists WHERE id=5;
+ROLLBACK TO SAVEPOINT s2;
+COMMIT;
+
+SELECT * FROM artists WHERE id IN (4, 5);
+
+SELECT citus.mitmproxy('conn.onQuery(query="^RELEASE").after(1).kill()');
+
+-- Release after rollback
+BEGIN;
+SAVEPOINT s1;
+ROLLBACK TO s1;
+RELEASE SAVEPOINT s1;
+SAVEPOINT s2;
+INSERT INTO artists VALUES (6, 'John J. Audubon');
+INSERT INTO artists VALUES (7, 'Emily Carr');
+ROLLBACK TO s2;
+RELEASE SAVEPOINT s2;
+COMMIT;
+
+SELECT * FROM artists WHERE id=7;
+
+SELECT citus.mitmproxy('conn.onQuery(query="^ROLLBACK").kill()');
+
+-- Recover from errors
+\set VERBOSITY terse
```

Comment:
this statement is not undone, is it expected to run the rest of the suite in this mode?